test/generate.test.ts

```diff
@@ -299,6 +299,35 @@ describe('generate', function () {
         });
       });
     });
+
+    describe('NxNxQ with x = N1 and y = N2', () => {
+      const specQ: SpecQuery = {
+        mark: SHORT_WILDCARD,
+        encodings: [{
+          channel: Channel.X,
+          field: 'N',
+          type: Type.NOMINAL
+        },{
+          channel: Channel.Y,
+          field: 'N20',
+          type: Type.NOMINAL
+        },{
+          channel: SHORT_WILDCARD,
+          field: 'Q',
+          type: Type.QUANTITATIVE
+        }]
+      };
+
+      const answerSet = generate(specQ, schema, CONFIG_WITH_AUTO_ADD_COUNT);
+      it('should generate a rect heatmap with color encoding the quantitative measure', () => {
+        answerSet.forEach((specM) => {
```

Comment:
Currently the test checks if the result contains rect with color. 
We may want to write another test that make sure we recommend rect with color over other cases.  

(This test is good, but just we may want to add more test.)

Reply:
Perhaps, we may need to add more test to the ranking part?

Reply:
Would these go in `mark.test.ts`? I see a Discrete-Discrete test there (where Rect > Point) but I'm not clear on how the tests in that file are able to handle 3rd dimensions like (with color Rect should be > Point but if the 3rd channel is unspecified then size is better and Point > Rect).